### PR TITLE
Fix CommandManager "locking/synchronizing" mechanism

### PR DESCRIPTION
**Pull request by njam**
`CM_Cli_CommandManager` has a mechanism to ensure a CLI command that defines the `@synchronized` annotation only runs once across the installation of the application.

See `monitorSynchronizedCommands`, `_lockCommand` and `_checkLock`.

The mechanism relies on a unique host identifier, by calling `hostid`, unfortunately that identifier is not unique across our hosts.

Let's investigate why `hostid` is not unique and/or find another mechanism.

Another improvement: in the foreach of `monitorSynchronizedCommands` let's only modify rows of the *current* host.

### CODE REVIEW

**Comment by njam** [comment]
@tomaszdurka please review

**Comment by njam** [comment]
@tomaszdurka please review

File: library/CM/Cli/CommandManager.php [hunk 1]
```diff
@@ -334,7 +334,19 @@ protected function _outputError($message) {
...
-        $file = new CM_File('/etc/machine-id'); // https://www.freedesktop.org/software/systemd/man/machine-id.html
+        // Global machine-id from systemd https://www.freedesktop.org/software/systemd/man/machine-id.html
+        $file = new CM_File('/etc/machine-id');
+
+        if (!$file->exists()) {
+            // Local machine-id as a backup
+            $serviceManager = CM_Service_Manager::getInstance();
+            $file = new CM_File('machine-id', $serviceManager->getFilesystems()->getData());
+            if (!$file->exists()) {
+                $uuid = Ramsey\Uuid\Uuid::uuid4()->toString();
```
Comment (tomaszdurka) [inline library/CM/Cli/CommandManager.php]: Well well if we create it like this, maybe we should not rely on the global id at all.

File: resources/config/deploy.php [hunk 2]
```diff
@@ -1,4 +1,4 @@
 <?php
 return function (CM_Config_Node $config) {
-    $config->deployVersion = 1;
+    $config->deployVersion = 1474973160;
```
Comment (tomaszdurka) [inline resources/config/deploy.php]: Intended?

File: library/CM/Cli/CommandManager.php (hunk 1 above)
Comment (njam) [inline library/CM/Cli/CommandManager.php]: The global id will be more reliable so I would still use it. In case of app re-installs etc?

**Comment by njam** [comment]
@tomaszdurka please review

File: resources/config/deploy.php (hunk 2 above)
Comment (tomaszdurka) [inline resources/config/deploy.php]: Is this intended? :)

Comment (njam) [inline resources/config/deploy.php]: It's an outdated diff. The reviews and the attached code can be a bit confusing :/ Check the PR's diff.

**Comment by njam** [comment]
@tomaszdurka please review

**Pull request merged**